### ISSUE TITLE
jax.lax.scatter_max returns infinite gradient with mode='clip' and valid clipped indices

### ISSUE THREAD

**Issue by ALinrunrun** [jax-ml/jax#40626]
### Description

`jax.lax.scatter_max` appears to return an incorrect infinite reverse-mode derivative when using `mode='clip'` with out-of-bounds indices that clip to valid positions.

In the example below, `mode='clip'` permits the out-of-bounds indices. The indices clip to distinct output positions, and neither `max` comparison is tied near `t = 0.03125`.

The scalar function is locally equivalent to:

```
2 * max(1, t + 1) - 3 + 3 * max(4, 4 - t)
```

At `t = 0.03125`, this local expression has derivative `2.0`. However, JAX returns `inf`.

### Minimal reproducible example

```
#!/usr/bin/env python3
import os
import sys
import math

os.environ.setdefault("CUDA_VISIBLE_DEVICES", "")
os.environ.setdefault("OMP_NUM_THREADS", "1")
os.environ.setdefault("JAX_PLATFORMS", "cpu")

try:
    import jax
    import jax.numpy as jnp
except ImportError as e:
    print(f"missing dep: {e}")
    sys.exit(2)

jax.config.update("jax_enable_x64", True)

def target(t):
    x = jnp.asarray([1, 3, 4], dtype=t.dtype)
    u = t * jnp.asarray([1, -1], dtype=t.dtype) + jnp.asarray([1, 4], dtype=t.dtype)
    i = jnp.asarray([[-3], [8]], dtype=jnp.int32)

    dnums = jax.lax.ScatterDimensionNumbers(
        update_window_dims=(),
        inserted_window_dims=(0,),
        scatter_dims_to_operand_dims=(0,),
    )

    y = jax.lax.scatter_max(
        x,
        i,
        u,
        dnums,
        indices_are_sorted=True,
        unique_indices=True,
        mode="clip",
    )

    return jnp.sum(y * jnp.asarray([2, -1, 3], dtype=t.dtype))

x = jnp.asarray(0.03125, dtype=jnp.float64)
expected = 2.0

actual = float(jax.grad(target)(x))

print("actual:", actual)
print("expected:", expected)

if not (math.isfinite(actual) and abs(actual - expected) <= 1e-12):
    sys.exit(0)
sys.exit(1)
```

### Actual result

```
actual: inf
expected: 2.0
```

### Expected result

```
2.0
```

The forward value is correct, but the reverse-mode derivative is not. The expected derivative follows from differentiating the local scalar expression:

```
2 * max(1, t + 1) - 3 + 3 * max(4, 4 - t)
```

At the tested point, the clipped scatter positions are valid and distinct, and the active `max` branches are not tied. JAX returns `inf` instead of the finite derivative `2.0`, suggesting that the reverse-mode rule for `scatter_max` with clipped indices is producing an invalid gradient contribution.

### System info (python version, jaxlib version, accelerator, etc.)

jax: 0.10.1
jaxlib: 0.10.1
numpy: 2.4.6
python: 3.13.13
platform: Ubuntu 24.04.3 LTS, Linux x86_64
accelerator: CPU backend

Note: GPU is present, but CUDA-enabled jaxlib is not installed, so JAX falls back to CPU.
